File: SDKs/amazonaws.com/cloudfront/2018-06-18/java/lib/src/main/java/openapisdk/models/shared/PublicKey.java
package openapisdk.models.shared;

import java.time.OffsetDateTime;

/**
 * PublicKey
 * A complex data type of public keys you add to CloudFront to use with features like field-level encryption.
**/
public class PublicKey {
    public OffsetDateTime createdTime;
    public PublicKey withCreatedTime(OffsetDateTime createdTime) {
        this.createdTime = createdTime;
        return this;
    }
    public String id;
    public PublicKey withId(String id) {
        this.id = id;
        return this;
    }
    public PublicKeyConfig publicKeyConfig;
    public PublicKey withPublicKeyConfig(PublicKeyConfig publicKeyConfig) {
        this.publicKeyConfig = publicKeyConfig;
        return this;
    }
}

File: SDKs/amazonaws.com/cloudfront/2019-03-26/python/src/sdk/models/shared/publickey.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from ..shared import publickeyconfig as shared_publickeyconfig


@dataclasses.dataclass
class PublicKey:
    r"""PublicKey
    A complex data type of public keys you add to CloudFront to use with features like field-level encryption.
    """
    
    created_time: datetime = dataclasses.field()
    id: str = dataclasses.field()
    public_key_config: shared_publickeyconfig.PublicKeyConfig = dataclasses.field()
    
